core/src/main/java/com/yahoo/imapnio/async/data/QResyncParameter.java

```diff
@@ -0,0 +1,70 @@
+package com.yahoo.imapnio.async.data;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class models the QRESYNC parameter defined in https://tools.ietf.org/html/rfc7162#page-26.
+ */
+public class QResyncParameter {
+    /** Last known UIDVALIDITY. */
+    private long uidValidity;
+
+    /** Last known modification sequence. */
+    private long modSeq;
+
+    /** Optional set of known UIDs. */
+    private MessageNumberSet[] knownUids;
+
+    /** Optional parenthesized list of known sequence ranges and their corresponding UIDs. */
+    private QResyncSeqMatchData seqMatchData;
+
+    /**
+     * Constructor.
+     *
+     * @param uidValidity last known uidvalidity
+     * @param modSeq last known modification sequence
+     * @param knownUids known UIDs
+     * @param seqMatchData known message sequence set and their corresponding UID
+     */
+    public QResyncParameter(final long uidValidity, final long modSeq, @Nullable final MessageNumberSet[] knownUids,
+                            @Nullable final QResyncSeqMatchData seqMatchData) {
+        this.uidValidity = uidValidity;
+        this.modSeq = modSeq;
+        if (knownUids != null) {
```

Comment:
what is the difference without if statement?

Reply:
no difference i can remove it.